### PR TITLE
[@types/hapi-pino] fix/improve types, improve tests

### PR DESCRIPTION
**Pull request by BlooJeans**
Summary:
- fix: `levelTags` -> `tags`
- fix: `tags` to allow for optional keys (before it required that you specified all keys)
- improve: add missing "fatal" from LogLevels (copied-directly from `@types/hapi`, since it's not exported from there)
- improve: updated `prettyPrint` and `redact` to include their advanced config options
- improve: add typedefs for `getChildBindings()` (new in 6.2.0)
- improve: more test coverage

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/pinojs/hapi-pino#options

- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/hapi-pino/index.d.ts [hunk 1]
```diff
@@ -22,22 +23,30 @@ declare module '@hapi/hapi' {
...
         logRouteTags?: boolean;
         stream?: NodeJS.WriteStream;
-        prettyPrint?: boolean;
-        levelTags?: { [key in LogLevels]: string };
-        allTags?: LogLevels;
-        serializers?: { [key: string]: (param: any) => void};
+        prettyPrint?: boolean | pino.PrettyOptions;
+        tags?: { [key in Level]?: string };
+        allTags?: Level;
+        serializers?: { [key: string]: SerializerFn };
```
Comment (saboya) [inline types/hapi-pino/index.d.ts]:
```suggestion
        serializers?: { [key: string]: pino.SerializerFn };
```

File: types/hapi-pino/index.d.ts [hunk 2]
```diff
@@ -22,22 +23,30 @@ declare module '@hapi/hapi' {
 }
 
 declare namespace HapiPino {
-    type LogLevels = 'trace' | 'debug' | 'info' | 'warn' | 'error';
+    type Level = 'fatal' | 'error' | 'warn' | 'info' | 'debug' | 'trace';
+    type SerializerFn = (value: any) => any;
```
Comment (saboya) [inline types/hapi-pino/index.d.ts]:
```suggestion
```

Comment (saboya) [reply types/hapi-pino/index.d.ts]: (assuming you're accepting the suggested changes to use `pino.SerializerFn`

**Review by saboya — CHANGES_REQUESTED** [review]
Thanks for improving the types. I made some suggestions to use more types from `pino` and to avoid definition duplication, but everything looks good.

File: types/hapi-pino/index.d.ts (hunk 1 above)
Comment (BlooJeans) [reply types/hapi-pino/index.d.ts]: oh nice catch - so `pino.SerializerFn` is actually exported, so therefore `pino.Level` is also.

**Comment by BlooJeans** [comment]
Good suggestions, PR is updated

**Review by saboya — APPROVED** [review]

File: types/hapi-pino/index.d.ts [hunk 3]
```diff
@@ -22,22 +23,27 @@ declare module '@hapi/hapi' {
...
-        serializers?: { [key: string]: (param: any) => void};
+        prettyPrint?: boolean | pino.PrettyOptions;
+        tags?: { [key in pino.Level]?: string };
+        allTags?: pino.Level;
+        serializers?: { [key: string]: pino.SerializerFn };
+        getChildBindings?: (
+            req: Request,
+        ) => {
+            level?: pino.Level | string;
+            serializers?: Options['serializers'];
```
Comment (SimonSchick) [inline types/hapi-pino/index.d.ts]: Considering moving this into a separate type.

**Review by SimonSchick — APPROVED** [review]

File: types/hapi-pino/index.d.ts (hunk 3 above)
Comment (BlooJeans) [reply types/hapi-pino/index.d.ts]: Done

**Comment by BlooJeans** [comment]
I went ahead and added the new option `logRequestStart: boolean` option that was just merged in to bring it up to current

**Pull request merged**